Add Unleash feature flags example

Tymek opened the pull request:
### Description

Experimental example of using Vercel Edge to resolve and cache feature flags.

I'm looing for feedback if making a call to another edge function is acceptable. In my example `/api/unleash-proxy-definitions` endpoint is there to cache the request.

### Demo URL

https://feature-flag-unleash.vercel.app

### Type of Change

- [x] New Example
- [ ] Example updates (Bug fixes, new features, etc.)
- [ ] Other (changes to the codebase, but not to examples)

### New Example Checklist

- [x] 🛫 ~~`npm run new-example`~~ `pnpm new-example` was used to create the example
- [ ] 📚 The template wasn't used but I carefuly read the [Adding a new example](https://github.com/vercel/examples#adding-a-new-example) steps and implemented them in the example
- [x] 📱 Is it responsive? Are mobile and tablets considered?

Tymek commented:
Can I please have some eyes on this? Thank you in advance.

@lfades @steven-tey @dominiksipowicz

leerob commented on edge-middleware/feature-flag-unleash/.env.example:
```diff
@@ -0,0 +1,5 @@
+UNLEASH_RELAY_SECRET=yourkey
+# UNLEASH_CACHE_CONTROL=public, s-maxage=1, stale-while-revalidate=59
+UNLEASH_SERVER_API_URL=https://app.unleash-hosted.com/demo/api
+UNLEASH_SERVER_API_TOKEN=test-server:default.8a090f30679be7254af997864d66b86e44dcfc5291916adff72a0fb5
```
Are these safe to include?

leerob commented on edge-middleware/feature-flag-unleash/README.md:
```diff
@@ -0,0 +1,52 @@
+
+# Feature Flags with Unleash
+
+Setting up [Unleash](https://www.getunleash.io/) Proxy on Vercel Edge middleware. It puts the load of getting feature flag definitions on Edge Network Cache, decreasing traffic to the Unleash instance. This example does not cover Static Generation, which is another way of scaling your application.
```
```suggestion
Use [Unleash](https://www.getunleash.io/) with Vercel Edge Middleware. This example puts the load of getting feature flag definitions on the Edge Network Cache, decreasing traffic to the Unleash instance.
```

Tymek commented:
Hi @leerob,

Just wanted to check in and see if there's any further feedback you'd like to provide. I've addressed the comments you made earlier and am hoping we can move forward with merging this. Thank you for your time and consideration.
